packages/web/src/components/welcome-modal/WelcomeModal.tsx

```diff
@@ -0,0 +1,107 @@
+import { useCallback } from 'react'
+
+import { fillString } from '@audius/common'
+import {
+  Button,
+  ButtonType,
+  Flex,
+  IconArrowRight,
+  Text,
+  IconCloudUpload,
+  Avatar
+} from '@audius/harmony'
+import { Modal } from '@audius/stems'
+import { push as pushRoute } from 'connected-react-router'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { useModalState } from 'common/hooks/useModalState'
+import {
+  getNameField,
+  getProfileImageField
+} from 'common/store/pages/signon/selectors'
+import Drawer from 'components/drawer/Drawer'
+import { isMobile as getIsMobile } from 'utils/clientUtil'
+import { TRENDING_PAGE, UPLOAD_PAGE } from 'utils/route'
+
+const messages = {
+  welcome: 'Welcome to Audius%0! 🎉',
+  startListening: 'Start Listening',
+  upload: 'Upload'
+}
+
+const isMobile = getIsMobile()
+const Container = isMobile ? Drawer : Modal
+
+export const WelcomeModal = () => {
+  const dispatch = useDispatch()
+  const { value: userName } = useSelector(getNameField)
+  const profileImage = useSelector(getProfileImageField)
+  const [isOpen, setIsOpen] = useModalState('Welcome')
+
+  const onClose = useCallback(() => {
+    setIsOpen(false)
+  }, [setIsOpen])
+
+  const handleClickListen = useCallback(() => {
+    dispatch(pushRoute(TRENDING_PAGE))
+    onClose()
+  }, [dispatch, onClose])
+
+  const handleClickUpload = useCallback(() => {
+    dispatch(pushRoute(UPLOAD_PAGE))
+    onClose()
+  }, [dispatch, onClose])
+
+  return (
+    <Container isOpen={isOpen} onClose={onClose} size='small'>
+      <Flex
+        h='96px'
+        justifyContent='center'
+        css={{
+          zIndex: 1,
+          backgroundColor: '#ddd',
+          background: `url("${profileImage?.url}")`,
+          ...(!isMobile && {
+            '::before': {
+              content: '""',
+              position: 'absolute',
+              top: 0,
+              left: 0,
+              height: '100%',
+              width: '100%',
+              backdropFilter: 'blur(10px)'
+            }
+          })
+        }}
+      >
+        <Avatar
+          src={profileImage?.url}
+          css={{ position: 'absolute', top: 40, height: 96, width: 96 }}
+        />
+      </Flex>
+      <Flex direction='column' p='xl' pt='3xl' gap='xl'>
+        <Flex direction='column' css={{ textAlign: 'center' }} gap='l'>
+          <Text variant='label' size='xl' strength='strong'>
+            {fillString(messages.welcome, userName ? `, ${userName}` : '')}
+          </Text>
+          <Text variant='body' size='l'>
+            You’re in! Discover music from our talented DJs, producers, and
+            artists.
+          </Text>
+        </Flex>
+        <Flex direction='column' gap='s'>
+          <Button iconRight={IconArrowRight} onClick={handleClickListen}>
```

Comment:
lets make both of these buttons links. so the way to do that would be 
```tsx
<Button ... asChild>
  <Link to={TRENDING}>{messages.startListening}</Link>
</Button
```

Where link is
```
import { Link } from 'react-router-dom'
```

This is extra nice because we get better a11y and it's easier to write cause you dont need click-handlers  + push.

Since we want to close the modal in both cases, we can still have an `onClick` for the button, but it would just be
```tsx
<Button onClick={onClose} ... />
```